### PR TITLE
[InstallAPI] Set InstallAPI as a standalone tool instead of CC1 action

### PR DESCRIPTION
**Pull request by cyndyishida**
Installapi has important distinctions when compared to the clang driver, so much that, it doesn't make much sense to try to integrate into it.

This patch partially reverts the CC1 action & driver support to replace with its own driver as a clang tool.

For distribution, we could use `LLVM_TOOL_LLVM_DRIVER_BUILD` mechanism for integrating the functionality into clang such that the toolchain size is less impacted.

### CODE REVIEW

File: clang/include/clang/InstallAPI/Context.h [hunk 1]
```diff
@@ -35,19 +36,19 @@ struct InstallAPIContext {
...
-  llvm::IntrusiveRefCntPtr<DiagnosticsEngine> Diags = nullptr;
-
   /// File Path of output location.
   StringRef OutputLoc{};
 
   /// What encoding to write output as.
   llvm::MachO::FileType FT = llvm::MachO::FileType::TBD_V5;
 };
 
+class InstallAPIAction : public ASTFrontendAction {
```
Comment (ributzka) [inline clang/include/clang/InstallAPI/Context.h:46]: I would move this part into a dedicated header file.

File: clang/tools/clang-installapi/ClangInstallAPI.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
+#include "Options.h"
```
Comment (ributzka) [inline clang/tools/clang-installapi/ClangInstallAPI.cpp:1]: The license header is missing.

File: clang/tools/clang-installapi/Options.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
+#include "Options.h"
```
Comment (ributzka) [inline clang/tools/clang-installapi/Options.cpp:1]: Please add the license/file header.

File: clang/include/clang/InstallAPI/Context.h (hunk 1 above)
Comment (cyndyishida) [reply clang/include/clang/InstallAPI/Context.h]: Removed this entirely for now. Will add back in follow-up where the header input processing + frontend are hooked together.

**Review by ributzka — APPROVED** [review]
LGTM

**Pull request merged**